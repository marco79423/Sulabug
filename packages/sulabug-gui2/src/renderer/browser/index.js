import React, {useEffect} from 'react'
import {useDispatch} from 'react-redux'
import {Redirect, Route, Switch} from 'react-router-dom'
import {createMuiTheme, ThemeProvider} from '@material-ui/core/styles'
import {lime, teal} from '@material-ui/core/colors'

import CollectionPage from './pages/CollectionPage'
import BrowsePage from './pages/BrowsePage'
import SettingsPage from './pages/SettingsPage'
import DownloadPage from './pages/DownloadPage'


export default function Browser() {
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch({type: 'common/initialize-app', data: 'browser'})
  }, [])

  const theme = createMuiTheme({
    typography: {
      h2: {
        fontSize: '1.6rem',
      },
      subtitle2: {
        fontSize: '0.8rem',
      },
      body2: {
        fontSize: '0.9rem',
      }
    },
    palette: {
      primary: {
        main: teal[800]
      },
      secondary: {
        main: lime[600]
      },
    },
  })

  return (
    <ThemeProvider theme={theme}>
      <Switch>
        <Redirect exact from='/browser' to='/browser/collection'/>
        <Route exact path='/browser/collection' component={CollectionPage}/>
        <Route exact path='/browser/browse' component={BrowsePage}/>
        <Route exact path='/browser/download' component={DownloadPage}/>
        <Route exact path='/browser/settings' component={SettingsPage}/>
      </Switch>
    </ThemeProvider>
  )
}
