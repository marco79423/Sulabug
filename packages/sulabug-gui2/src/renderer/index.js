import React from 'react'
import * as ReactDOM from 'react-dom'
import {HashRouter as Router, Route, Switch} from 'react-router-dom'
import {Redirect} from 'react-router'
import {Provider} from 'react-redux'

import configureStore from './configureStore'
import Browser from './browser'

const store = configureStore()

function App() {
  return (
    <Provider store={store}>
      <Router>
        <Switch>
          <Redirect exact from='/' to='/browser'/>
          <Route path="/browser" component={Browser}/>
        </Switch>
      </Router>
    </Provider>
  )
}

ReactDOM.render(
  <App/>,
  document.getElementById('app'),
)
