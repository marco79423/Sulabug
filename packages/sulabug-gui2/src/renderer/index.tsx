import React, {lazy, Suspense} from 'react'
import * as ReactDOM from 'react-dom'
import {HashRouter, Route, Switch} from 'react-router-dom'
import {Redirect} from 'react-router'

const Manager = lazy(() => import('./manager'))
const Browser = lazy(() => import('./browser'))

function App() {
  return (
    <HashRouter>
      <Suspense fallback={<div>loading</div>}>
        <Switch>
          <Redirect exact from='/' to='/manager'/>
          <Route path="/manager" component={Manager}/>
          <Route path="/browser" component={Browser}/>
        </Switch>
      </Suspense>
    </HashRouter>
  )
}

ReactDOM.render(
  <App/>,
  document.getElementById('app'),
)
