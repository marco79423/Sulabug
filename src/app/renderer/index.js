import * as React from 'react'
import ReactDOM from 'react-dom'

import 'reflect-metadata'

import App from '../../imports/app/App'

ReactDOM.render(
  <App/>,
  document.getElementById('app')
)
