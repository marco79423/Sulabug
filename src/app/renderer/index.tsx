import * as React from 'react'
import {render} from 'react-dom'
import App from '../../imports/ui/pages/App'

render(
  <App/>,
  document.getElementById('app')
)