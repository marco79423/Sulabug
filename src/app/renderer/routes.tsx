import React from 'react'
import { Router, Route, Switch } from 'react-router'
import createBrowserHistory from 'history/createBrowserHistory'

import BrowsePage from '../../imports/ui/pages/App'

const browserHistory = createBrowserHistory()

export const renderRoutes = () => (
  <Router history={browserHistory}>
    <Switch>
      <Route exact path="/" component={BrowsePage}/>
    </Switch>
  </Router>
)
