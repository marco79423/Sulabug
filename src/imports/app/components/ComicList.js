import * as React from 'react'
import {createStyles, withStyles} from '@material-ui/core/styles'
import LinearProgress from '@material-ui/core/LinearProgress/LinearProgress'
import List from '@material-ui/core/List'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails'
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary'
import ExpandMoreIcon from '@material-ui/icons/ExpandMore'
import ExpansionPanel from '@material-ui/core/ExpansionPanel'
import Divider from '@material-ui/core/Divider'
import ExpansionPanelActions from '@material-ui/core/ExpansionPanelActions'
import dateFormat from 'dateformat'


import ComicListItem from './ComicListItem'

const styles = (theme) => createStyles({
  root: {
    padding: 0,
  },

  column: {
    flexBasis: '40%',
  },
})


export class ComicList extends React.Component {

  renderMetaSection = () => {
    const {classes, userProfile} = this.props

    const updatedTimeStr = userProfile.databaseUpdatedTime ? dateFormat(userProfile.databaseUpdatedTime, 'yyyy/mm/dd HH:MM') : '從未更新'

    return (
      <ExpansionPanel>
        <ExpansionPanelSummary expandIcon={<ExpandMoreIcon/>}>
          <div className={classes.column}/>
          <div>
            <Typography inline>更新時間： {updatedTimeStr}</Typography>
          </div>
        </ExpansionPanelSummary>
        <ExpansionPanelDetails>
          <Typography>
            漫畫資料庫每天會自動更新一次，如果收藏的漫畫內容有更新會自動下載並通知。
          </Typography>
        </ExpansionPanelDetails>
        <Divider/>
        <ExpansionPanelActions>
          <Button variant="contained" onClick={this.props.updateComicDatabase}>手動更新資料庫</Button>
        </ExpansionPanelActions>
      </ExpansionPanel>
    )
  }

  render() {
    const {classes, loading, comics, collection, addComicToCollection} = this.props
    if (loading) {
      return <LinearProgress color="secondary" variant="query"/>
    }
    return (
      <>
        {this.renderMetaSection()}
        <List className={classes.root}>
          {comics.map((comic) => <ComicListItem key={comic.id} comic={comic} collection={collection}
                                                addComicToCollection={addComicToCollection}/>)}
        </List>
      </>
    )
  }
}

export default withStyles(styles)(ComicList)
