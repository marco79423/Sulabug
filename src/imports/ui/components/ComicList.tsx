import * as React from 'react'
import {createStyles, Theme, withStyles} from '@material-ui/core/styles'
import List from '@material-ui/core/List'

import ComicListItem from './ComicListItem'

const styles = (theme: Theme) => createStyles({
  root: {
    padding: 0,
  },
})

interface PropsTypes {
  classes: any,
  comics: any,
}

export class ComicList extends React.Component<PropsTypes, {}> {
  render() {
    const {classes, comics} = this.props

    return (
      <List className={classes.root}>
        {comics.map((comic: any) => <ComicListItem key={comic.id} comic={comic}/>)}
      </List>
    )
  }
}

export default withStyles(styles)(ComicList)
