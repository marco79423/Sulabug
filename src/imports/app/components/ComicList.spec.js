import 'reflect-metadata'

import React from 'react'
import renderer from 'react-test-renderer'

import ComicList from './ComicList'


describe('ComicList', () => {
  it('will render comic list correctly', async () => {
    const component = renderer.create(
      <ComicList loading={false}
                 comics={[]}
                 createDownloadTask={() => {
                 }}/>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('will render comic list correctly in loading state', async () => {
    const component = renderer.create(
      <ComicList loading={true}
                 comics={[]}
                 createDownloadTask={() => {
                 }}/>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
