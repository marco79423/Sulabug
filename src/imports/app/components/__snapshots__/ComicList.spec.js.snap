// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicList will render comic list correctly 1`] = `
<ul
  className="MuiList-root-2 MuiList-padding-3 ComicList-root-1"
/>
`;

exports[`ComicList will render comic list correctly in loading state 1`] = `
<div
  className="MuiLinearProgress-root-6 MuiLinearProgress-colorSecondary-8 MuiLinearProgress-query-12"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar-16 MuiLinearProgress-barColorSecondary-18 MuiLinearProgress-bar1Indeterminate-19"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar-16 MuiLinearProgress-barColorSecondary-18 MuiLinearProgress-bar2Indeterminate-22"
    style={Object {}}
  />
</div>
`;
